src/main/java/com/couchbase/lite/replicator/BulkDownloader.java

```diff
@@ -170,8 +171,17 @@ protected void executeRequest(HttpClient httpClient, HttpUriRequest request) {
                 else {
                     Log.v(Log.TAG_SYNC, "contentTypeHeader is not multipart = %s",contentTypeHeader.getValue());
                     if (entity != null) {
+
+                        inputStream = entity.getContent();
+
+                        // decompress if contentEncoding is gzip
+                        Header contentEncoding = entity.getContentEncoding();
+                        if(contentEncoding != null && contentEncoding.getValue().contains("gzip")){
```

Comment:
Can you factor this:

```
if(contentEncoding != null && contentEncoding.getValue().contains("gzip")){
```

into an isGzip(contentEncodingHeader) method?

Also this code below can be changed to call that same method:

```
    protected void startJSONBufferWithHeaders(Map<String, String> headers){
        jsonBuffer = new ByteArrayBuffer(1024);
        String contentEncoding = headers.get("Content-Encoding");
        jsonCompressed = contentEncoding != null &&  contentEncoding.indexOf("gzip") >= 0;
    }
```